test/parallel/test-stringbytes-external.js

```diff
@@ -126,50 +122,57 @@ var PRE_3OF4_APEX = Math.ceil((EXTERN_APEX / 4) * 3) - RADIOS;
     return;
   }
 
+  const buf0 = new Buffer(kStringMaxLength * 2 + 2);
```

Comment:
That would work. my worry is that we allocate 256MB more than needed above, and if we hang on to that then the tests that turn these into strings will need to increase the heap by an additional 256MB. Hence why over allocating in the check above is a small way to ensure the heap has enough space for the `toString()` calls below.